lib/spec/router.sw-spec.js

```diff
@@ -206,4 +206,142 @@ describe('Router instances', function () {
+        var mw2 = function () {},
+            mw3 = function () {};
+
+        router.get('/pre/*/fix/*', mw);
+        router.get('*/suf/*/fix', mw2);
+        router.get('*inner*', mw3);
+
+        expect(router.match('get', '/')).to.be.empty;
+        expect(router.match('get', '/pre/123/fix/')).to.deep.equal([mw]);
+        expect(router.match('get', '/pre/123/fix/456')).to.deep.equal([mw]);
+        expect(router.match('get', '/pre/123/')).to.be.empty;
+
+        expect(router.match('get', '/matched/suf/123/fix')).to.deep.equal([mw2]);
+        expect(router.match('get', '/any/other/suf/123/456/fix')).to.deep.equal([mw2]);
+        expect(router.match('get', '/suf/nopre/fix')).to.deep.equal([mw2]);
+        expect(router.match('get', '/fix')).to.be.empty;
+
+        expect(router.match('get', '/pre/suf/fix/')).to.deep.equal([mw]);
+
+        expect(router.match('get', '/inner')).to.deep.equal([mw3]);
+        expect(router.match('get', '/beginners')).to.deep.equal([mw3]);
+        expect(router.match('get', '/pre/inner/fix/')).to.deep.equal([mw, mw3]);
+        expect(router.match('get', '/inner/suf/other/fix/')).to.deep.equal([mw2, mw3]);
+
+        expect(router.match('get', '/pre/123/fix/inner/suf/456/fix/')).to.deep.equal([mw, mw2, mw3]);
+        expect(router.match('get', '/pre/suf/inner/fix/')).to.deep.equal([mw, mw2, mw3]);
+      });
+
+      it('named placeholders (:foo) should match similar to anonymous ones', function() {
+        router.get('/:foo', mw);
+        expect(router.match('get', '/foo')).to.deep.equal([mw]);
+        expect(router.match('get', '/foo/bar')).to.deep.equal([mw]);
+      });
+
+      it('named placeholders (:foo) must match at least one character (no empty matches allowed)', function() {
+        router.get('/:foo', mw);
+        expect(router.match('get', '/')).to.be.empty;
+        expect(router.match('get', '/foo/bar')).to.deep.equal([mw]);
+      });
+
+      it('placeholder crowding for anonymous placeholder (**) should be treated same as a single placeholder', function() {
+        var mw2 = function () {};
+
+        expect(function() { router.get('/f**bar', mw); }).to.not.throw;
+        router.get('/f*bar', mw2);
+
+        expect(router.match('get', '/foo')).to.deep.equal(router.match('get', '/foo'));
+        expect(router.match('get', '/foobar')).to.deep.equal(router.match('get', '/foobar'));
+        expect(router.match('get', '/foobabarbar')).to.deep.equal(router.match('get', '/foobabarbar'));
+        expect(router.match('get', '/foo/bar/baz')).to.deep.equal(router.match('get', '/foo/bar/baz'));
+        expect(router.match('get', '/foo/doh/bar')).to.deep.equal(router.match('get', '/foo/doh/bar'));
+      });
+
+      it('placeholder crowding for named placeholder (:foo:bar) should throw', function() {
+        expect(function () { router.get('/foo/:bar:baz', mw); }).to.throw(Error, 'Invalid usage of named placeholders');
+      });
+
```

Comment:
Add a couple of more tests to test _triple_ crowding: `***` and `:foor:bar:foo`.